apis/v1alpha1/httproute_types.go

```diff
@@ -167,6 +167,17 @@ type HTTPRouteRule struct {
 	// path match on "/", which has the effect of matching every
 	// HTTP request.
 	//
+	//
+	// A client request may match multiple HTTP route rules,
+	// to differentiate we define matching precedence where matches
+	// MUST be determined in order of the following criteria, continuing on ties:
+	// * longest matching combination of hostname and path.
```

Comment:
we can probably split this up so longest hostname match is considered first, then path.